<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>276</width>
    <height>1339</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BIMbots</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="labelLogo">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="font">
      <font>
       <pointsize>16</pointsize>
      </font>
     </property>
     <property name="text">
      <string>BIMbots logo</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupProgress">
     <property name="title">
      <string/>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QProgressBar" name="progressBar">
        <property name="value">
         <number>10</number>
        </property>
        <property name="format">
         <string>connecting...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonCancelProgress">
        <property name="toolTip">
         <string>Cancel the current connection</string>
        </property>
        <property name="text">
         <string>Cancel</string>
        </property>
        <property name="icon">
         <iconset theme="process-stop">
          <normaloff/>
         </iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Available services</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QTreeWidget" name="servicesList">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <attribute name="headerVisible">
         <bool>false</bool>
        </attribute>
        <column>
         <property name="text">
          <string notr="true">1</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonRescan">
        <property name="toolTip">
         <string>Tries to obtain a list of available services from all registered servers</string>
        </property>
        <property name="text">
         <string>Rescan services</string>
        </property>
        <property name="icon">
         <iconset theme="edit-redo">
          <normaloff/>
         </iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupRescan">
        <property name="title">
         <string>Service discovery options:</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_8">
         <item>
          <widget class="QCheckBox" name="checkAutoDiscover">
           <property name="text">
            <string>Auto-discover new services</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="checkShowUnreachable">
           <property name="text">
            <string>Show unreachable providers</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboFilter">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <item>
            <property name="text">
             <string>Filter by category...</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="layoutPrice">
           <item>
            <widget class="QLabel" name="labelPrice">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>max price:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEditPrice">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="placeholderText">
              <string>0 USD</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="layoutRescanButtons">
           <item>
            <widget class="QPushButton" name="buttonDoRescan">
             <property name="text">
              <string>Scan</string>
             </property>
             <property name="icon">
              <iconset theme="edit-redo">
               <normaloff/>
              </iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="buttonCancelRescan">
             <property name="text">
              <string>Cancel</string>
             </property>
             <property name="icon">
              <iconset theme="cancel">
               <normaloff/>
              </iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonAddService">
        <property name="toolTip">
         <string>Adds a new entry to the list of known service providers</string>
        </property>
        <property name="text">
         <string>Add custom service provider</string>
        </property>
        <property name="icon">
         <iconset theme="list-add">
          <normaloff/>
         </iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupAddService">
        <property name="title">
         <string>Add custom service provider:</string>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QGridLayout" name="layoutAddService">
           <item row="0" column="0">
            <widget class="QLabel" name="labelServiceName">
             <property name="text">
              <string>Name:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="labelServiceUrl">
             <property name="text">
              <string>URL:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="lineEditServiceName">
             <property name="toolTip">
              <string>A custom name to identify this services provider</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="lineEditServiceUrl">
             <property name="toolTip">
              <string>The URL of this services provider</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="layoutServiceButtons">
           <item>
            <widget class="QPushButton" name="buttonSaveService">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Add this server to the list of known service providers</string>
             </property>
             <property name="text">
              <string>Add</string>
             </property>
             <property name="icon">
              <iconset theme="list-add">
               <normaloff/>
              </iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="buttonCancelService">
             <property name="toolTip">
              <string>Cancels the operation</string>
             </property>
             <property name="text">
              <string>Cancel</string>
             </property>
             <property name="icon">
              <iconset theme="cancel">
               <normaloff/>
              </iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonRemoveService">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Removes a selected service from the list above</string>
        </property>
        <property name="text">
         <string>Remove custom service provider</string>
        </property>
        <property name="icon">
         <iconset theme="remove">
          <normaloff/>
         </iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonAuthenticate">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Authenticates with a selected service. Opens a web browser window.</string>
        </property>
        <property name="text">
         <string>Authenticate</string>
        </property>
        <property name="icon">
         <iconset theme="user">
          <normaloff/>
         </iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupAuthenticate">
        <property name="title">
         <string>Paste authentication credentials here:</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <layout class="QGridLayout" name="layoutAuth">
           <item row="1" column="0">
            <widget class="QLabel" name="labelAuthenticateUrl">
             <property name="text">
              <string>Service URL:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="lineEditAuthenticateUrl">
             <property name="toolTip">
              <string>The service URL returned by the authentication process</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="labelAuthenticateToken">
             <property name="text">
              <string>Token:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="lineEditAuthenticateToken">
             <property name="toolTip">
              <string>The token returned by the authentication process</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="layoutAuthButtons">
           <item>
            <widget class="QPushButton" name="buttonSaveAuthenticate">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Saves the authentication credentials</string>
             </property>
             <property name="text">
              <string>Save</string>
             </property>
             <property name="icon">
              <iconset theme="document-save">
               <normaloff/>
              </iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="buttonCancelAuthenticate">
             <property name="text">
              <string>Cancel</string>
             </property>
             <property name="icon">
              <iconset theme="cancel">
               <normaloff/>
              </iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupRun">
     <property name="title">
      <string>Run selected service</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QLabel" name="labelRun">
        <property name="text">
         <string>Run service on:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="scopeList">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>80</height>
         </size>
        </property>
        <item>
         <property name="text">
          <string>All document objects</string>
         </property>
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Sends all the objects contained in the active document to the selected service. Warning, all the inner object components wil be sent too, this is usually not what you want. Use for testing purposes only.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>All visible objects</string>
         </property>
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Sends all the visible objects to the selected service. All the needed components will be gathered automatically. Choose this option if your model doesn't use IFC building structures such as Sites, Buildings or Storeys&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Selected objects</string>
         </property>
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Sends the selected objects to the given service. All the necessary components will be gathered authomatically. You can select only one or more groups or sites, buildings or storeys, all their child objects will be collected too. This is usually the best option. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Test payload</string>
         </property>
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Sends a very simple test payload IFC to the service, that is guaranteed to work. Use this to test if the service works correctly.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="buttonRun">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Run service</string>
        </property>
        <property name="icon">
         <iconset theme="media-playback-start">
          <normaloff/>
         </iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupResults">
     <property name="title">
      <string>Results</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QTreeWidget" name="treeResults">
        <attribute name="headerVisible">
         <bool>false</bool>
        </attribute>
        <column>
         <property name="text">
          <string notr="true">1</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
